### DISCUSSION TITLE
about Peridynamic fatigue model

### DISCUSSION THREAD

**Discussion by Lumos1110** [PeriHub/PeriLab.jl#146] [General]
I would like to add a fatigue model with the following literature reference:DOI:[10.2172/1160289](https://doi.org/10.2172/1160289)
I have a question at the moment: we need to apply a sinusoidal load in a loop to calculate the strain in each bond when the maximum load is reached. However, there is a problem that for total time, the calculated key strains are different for different total times. @CWillberg @JTHesse

**Comment by JTHesse** [comment]
Could you maybe share your .yaml input file? Then we can take a look.

**Reply by Lumos1110** [reply]
I just did a simple writeup with matlab code, I'll send it to you if needed    : D

**Reply by JTHesse** [reply]
So your question is how to do this in PeriLab?
Sinusoidal load can be applied directly in the BC:
```
 BC_1:
      Coordinate: y
      Node Set: Node Set 1
      Variable: Force Densities
      Type: Dirichlet
      Value: a * sin(b * t + c) + d
```

**Reply by Lumos1110** [reply]
This I am aware of, but it creates a problem.
You could make a plane with the left side fixed and the right side stretched, using a quasi-static algorithm. dt=1s, then apply a sinusoidal load to the right side with stress ratio=0, and then determine the elongation of each bond. You will find that when the total time is taken 100s and 1000s their elongation is different.

**Reply by JTHesse** [reply]
Ok, so this is a general peridynamic question or is this somehow related to PeriLab?
If you would like to implement the functionalities mentioned in the Paper into PeriLab, go ahead and open a [feature request](https://github.com/PeriHub/PeriLab.jl/issues/new/choose).
There could be many reasons why you are seeing this behavior, especially with a self-made matlab code, but until we don't have a running example, it's hard to say.
I'm leaving this discussion open for now, maybe someone else can comment on your question.

**Reply by Lumos1110** [reply]
Ok, I've found the problem, my ultimate goal is to couple corrosion and fatigue in perilab, so I'd like to implement it step by step. 
I just rewrote the implementation in perilab as you said and it's great!

**Reply by JTHesse** [reply]
Perfect, if you are ready and the tests are running add a pull request ;)
